tests/integration/test_utils/data_source_config/base.py

```diff
@@ -53,6 +55,33 @@ def test_id(self) -> str:
 
         return "-".join(non_null_parts)
 
+    @override
+    def __eq__(self, value: object) -> bool:
+        if not isinstance(value, DataSourceTestConfig):
+            return False
+        return all(
+            [
+                super().__eq__(value),
+                self.label == value.label,
+                self.pytest_mark == value.pytest_mark,
+            ]
+        )
+
+    @override
+    def __hash__(self) -> int:
+        hashable_col_types = dict_to_tuple(self.column_types) if self.column_types else None
```

Comment:
If 2 objects have the same value of `__eq__` are they guaranteed to have the same hash? This seems to be a requirement of hash (eg so putting them in dictionaries works as expected): https://docs.python.org/3/reference/datamodel.html#object.__hash__

Reply:
>If 2 objects have the same value of __eq__ are they guaranteed to have the same hash

Reply:
I don't know if we are talking about the same thing? Using tuples is fine. I'm asking if `a == b` are we guaranteed that `a.__hash__() == b.__hash__()`?  Maybe it's true but it wasn't apparent when I read this code.

My link didn't work because the last characters got removed from the url (fixed now) but I'm referring to this from docs:
> The only required property is that objects which compare equal have the same hash value; it is advised to mix together the hash values of the components of the object that also play a part in comparison of objects by packing them into a tuple and hashing the tuple.

If we use this for a key in a dict, and this isn't true, the lookup can fail since both hash and equality are used in the lookup.

Reply:
Oh yeah, I misunderstood your question. My intention was definitely the implement this so if a == b are we guaranteed that a.__hash__() == b.__hash__(). I'm pretty confident it does this. But LMK if you aren't convinced (or if I'm still missing something 😄 )